From 3ab579f7266876a9bba6170b29dce02a7f1f85b7 Mon Sep 17 00:00:00 2001
From: shaoyunl <Shaoyun.Liu@amd.com>
Date: Mon, 4 Apr 2016 16:35:03 -0400
Subject: [PATCH 1760/2094] drm/amdgpu: Add PDE0 vm fault as invalid address
 protection fault

The VM generally raises two kind of invalid address protection fault:
PDE0 (page directory entry not present) or VALID(page table entry
not valid). Both need to be reported as invalid protection fault to user.

Change-Id: I012ff65e4c9def01534897887cafdc127b7c07d1
Signed-off-by: shaoyun liu <shaoyun.liu@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/gmc_v7_0.c | 2 +-
 drivers/gpu/drm/amd/amdgpu/gmc_v8_0.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/gmc_v7_0.c b/drivers/gpu/drm/amd/amdgpu/gmc_v7_0.c
index 8633c0b..1f21616 100644
--- a/drivers/gpu/drm/amd/amdgpu/gmc_v7_0.c
+++ b/drivers/gpu/drm/amd/amdgpu/gmc_v7_0.c
@@ -835,7 +835,7 @@ static void gmc_v7_0_vm_decode_fault(struct amdgpu_device *adev,
 		info->vmid = vmid;
 		info->mc_id = mc_id;
 		info->page_addr = addr;
-		info->prot_valid = protections & 0x4 ? true : false;
+		info->prot_valid = protections & 0x6 ? true : false;
 		info->prot_read = protections & 0x8 ? true : false;
 		info->prot_write = protections & 0x10 ? true : false;
 		info->prot_exec = protections & 0x20 ? true : false;
diff --git a/drivers/gpu/drm/amd/amdgpu/gmc_v8_0.c b/drivers/gpu/drm/amd/amdgpu/gmc_v8_0.c
index 395713f..2c51671d 100644
--- a/drivers/gpu/drm/amd/amdgpu/gmc_v8_0.c
+++ b/drivers/gpu/drm/amd/amdgpu/gmc_v8_0.c
@@ -1038,7 +1038,7 @@ static void gmc_v8_0_vm_decode_fault(struct amdgpu_device *adev,
 		info->vmid = vmid;
 		info->mc_id = mc_id;
 		info->page_addr = addr;
-		info->prot_valid = protections & 0x4 ? true : false;
+		info->prot_valid = protections & 0x6 ? true : false;
 		info->prot_read = protections & 0x8 ? true : false;
 		info->prot_write = protections & 0x10 ? true : false;
 		info->prot_exec = protections & 0x20 ? true : false;
-- 
2.7.4

